### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by cmolinan**
### ACTIVITIES

> On exit from the app, it saves all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. rentals.json)
> On startup of the app, it loads all data from files, handling the case when one or more of the files are missing.

### CODE REVIEW

File: src/main.rb [hunk 1]
```diff
@@ -22,29 +23,38 @@ def main(db_data)
...
+    when '99'
+      storage.save_all(db_data)
+    when '100'
+      storage.restore_all(db_data)
+    else
+      puts "\nSelection not valid"
+    end
   end
-  main(db_data)
+  # main(db_data)
```
Comment (Nemwel-Boniface) [inline src/main.rb]: - [x] Kindly remove all commented out, un used. unwanted pieces of code from your code base. It makes it look messy and cluttered.

File: src/only_lists.rb [hunk 2]
```diff
@@ -24,7 +24,8 @@ def list_rentals(db_persons)
     print "\nID of the person: "
     person_id = gets.chomp.to_i
     person_fetch = db_persons.select { |person| person.id == person_id }
-    if person_fetch[0].id
+    # if person_fetch[0] && person_fetch[0].id
```
Comment (Nemwel-Boniface) [inline src/only_lists.rb]: - [x] Kindly remove all commented out, un used. unwanted pieces of code from your code base. It makes it look messy and cluttered.

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @cmolinan  and @Efiamotu-1    :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

- [x] For some reason, when I start the application, I am getting an error. The application seems to be starting but then gets stopped by the error. This has hindered me from testing out your project. Kindly look into why this is happening.

image


_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/main.rb (hunk 1 above)
Comment (cmolinan) [reply src/main.rb]: @Nemwel-Boniface Ok. I did it !

File: src/only_lists.rb (hunk 2 above)
Comment (cmolinan) [reply src/only_lists.rb]: @Nemwel-Boniface Done !

**Comment by cmolinan** [comment]
@Nemwel-Boniface  

image

The app did not run because, according to the instructions in the Readme file, the user must enter in the src folder, and then run: ruby main.rb
Just in case, we have improved that detail in the Readme file.

**Comment by Nemwel-Boniface** [comment]
Hello, @cmolinan I did make sure to include it in the screenshot I did ruby src/main.rb which is more the same as first cd to src then ruby main.rb. Good job however updating the README. Hope it runs smoothly now!

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @cmolinan, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by cmolinan** [comment]
> Hello, @cmolinan I did make sure to include it in the screenshot I did ruby src/main.rb which is more the same as first cd to src then ruby main.rb. Good job however updating the README. Hope it runs smoothly now!

@Nemwel-Boniface Thanks for your comment and review !!

**Comment by cmolinan** [comment]
@Mihndim2020 Thanks !!

**Pull request merged**